catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -517,6 +520,164 @@ public void init() {
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            //Test if parent metacard ID is valid and comment is not empty
+            if (!isVirtual) {
+                try {
+                    util.getMetacard(parentID, true);
+                } catch (NotFoundException e) {
+                    LOGGER.debug("Parent metacard ID is invalid!");
+                    throw new IllegalStateException("Parent metacard ID is invalid!");
+                }
+            }
+            if (comment.equals("")) {
+                LOGGER.info(util.getResponseWrapper("error", "No comment!"));
+                return util.getResponseWrapper("error", "No comment!");
+            }
+            MetacardType noteType = null;
+            for (MetacardType type : types) {
+                if (type.getName()
+                        .equals("resource-note")) {
+                    noteType = type;
+                    break;
+                }
+            }
+            if (noteType == null) {
+                LOGGER.debug("Metacard type 'note' has not been registered in system");
+                throw new IllegalStateException(
+                        "Metacard type 'note' has not been registered in system");
+            }
+            Metacard note = saveMetacard(new NoteMetacard(noteType, parentID, user, comment));
+            String getResponse = util.getResponseNoteJson(note);
+            if (getResponse == null) {
+                LOGGER.debug("An error occurred while storing the note");
+                throw new IllegalStateException("An error occurred while storing the note");
+            }
+            return util.getResponseWrapper("success", getResponse);
+        });
+
+        /*
+         * Notes may have a virtual parent, which means the parent is not directly accessible to
+         * the framework. For example, the parent may be stored as a text field inside another
+         * metacard. For non-virtual parents, the parent must be retrievable by the framework.
+         * If the call does not set the isVirtual flag, then it defaults to 'false'.
+         */
+        get("/notes/:parentid", (req, res) -> {
+            LOGGER.trace("Hit notes GET method");
+            String parentID = req.params(":parentid");
+            String[] isVirtualParams = req.queryMap()
+                    .toMap()
+                    .getOrDefault("isVirtual", new String[] {"false"});
+            Boolean isVirtual =
+                    isVirtualParams.length > 0 && Boolean.parseBoolean(isVirtualParams[0]);
+            if (!isVirtual) {
+                //Test if parent metacard ID is valid
+                try {
+                    util.getMetacard(parentID, true);
+                } catch (NotFoundException e) {
+                    LOGGER.debug("Parent metacard ID is invalid!");
```

Comment:
I think this line is OBE, but in the two places where a NFE is caught, I added a debug log message with the ID.